### PR TITLE
Launchpad support

### PR DESCRIPTION
**Pull request by lefty71**
I added some improvements to supporting launchpad via I2C

### CODE REVIEW

**Comment by jellea** [comment]
Awesome! Will review tonight! How are you testing this?

Since we're bumping the automat version:
1. Can we change the i2c address to 0xDA? We forgot to change it previously as it's now the same as the [TeleXo](https://store.bpcmusic.com/products/telexo).
2. Can/shall we add a bypass the midi note configuration for i2c events? So sending to output 1 always sends to output 1?

**Comment by lefty71** [comment]
I have a different step sequencer that I am using as a clock source for the Automat.

1.   Not a problem

2. That’s how the current code within the LAUNCHPAD_SUPPORT in automat.ino works.

File: automat/automat.ino [hunk 1]
```diff
@@ -14,7 +14,7 @@
 // i2c constants
 // TODO: this is the temporary i2c address same as the TELEXo Teletype Expander,
```
Comment (jellea) [inline automat/automat.ino]: Let's remove this todo?

File: automat/automat.ino [hunk 2]
```diff
@@ -428,6 +518,15 @@ void receiveI2CEvent(int len)
         uint8_t byte2 = Wire.read();
         uint8_t byte3 = Wire.read();
 
+#if LAUNCHPAD_SUPPORT
+        char pin = byte2 - 1;
+        char velocity = byte3;
+        if (velocity > 0) {
+          handleNoteOn(pin, velocity);
```
Comment (jellea) [inline automat/automat.ino:523]: Great! I missed this part!

File: automat/automatConstants.h [hunk 3]
```diff
@@ -35,8 +35,9 @@
 
 
 #define AUTOMAT_MINI 0
+#define LAUNCHPAD_SUPPORT 1
```
Comment (jellea) [inline automat/automatConstants.h]: Just out of curiousity: Why you add a flag for this?

File: automat/dadaSysEx.hpp [hunk 4]
```diff
@@ -68,8 +89,10 @@ bool dadaSysEx::handleSysEx(byte * arr, unsigned len)
    }
 
    if (*arr++ != 0) {
+     SYSEX_DEBUG("Sysex bad delimiter");
```
Comment (jellea) [inline automat/dadaSysEx.hpp:92]: Nice! Thanks for all the added debugging!

File: automat/automatConstants.h (hunk 3 above)
Comment (jellea) [reply automat/automatConstants.h]: Also, how about calling it `SIS_SUPPORT`, because it's not launchpad dependent :)

**Review by jellea — COMMENTED** [review]
Amazing! Love that you can use it in combination with a DAW now! Couple of questions and suggestions :)

File: automat/automat.ino (hunk 1 above)
Comment (lefty71) [reply automat/automat.ino]: Done

File: automat/automatConstants.h (hunk 3 above)
Comment (lefty71) [reply automat/automatConstants.h]: It disables some of the older I2C code which relies on midi notes matching the automat note configuration.   i will call it SIS_SUPPORT for now, or I could just remove the older code.

Comment (jellea) [reply automat/automatConstants.h]: Let's remove the older code to keep things simple!

Comment (lefty71) [reply automat/automatConstants.h]: I removed the older code.   I left the SIS_SUPPORT define here as it helps document the code.

Comment (jellea) [reply automat/automatConstants.h]: Perfect! Let's ship it!

**Review by jellea — APPROVED** [review]
🙌  Excited to get this in!

**Pull request merged**